py3status/modules/imap.py

```diff
@@ -325,19 +360,58 @@ def _get_mail_count(self):
                         auth_string = "user={}\1auth=Bearer {}\1\1".format(
                             self.user, self.creds.token
                         )
-                        self.connection.authenticate("XOAUTH2", lambda x: auth_string)
+                        self.connection.authenticate(
+                            "XOAUTH2", lambda x: auth_string
+                        )
                     else:
                         # Login with user and password
                         self.connection.login(self.user, self.password)
 
                 tmp_mail_count = 0
-                directories = self.mailbox.split(",")
+                if self.mailbox:
+                    directories = self.mailbox.split(",")
+                else:
+                    try:
+                        directories
+                    except:
+                        directories = []
+                        filters = "|".join(self.filter)
+                        objs = [x.decode() for x in self.connection.list()[1]]
+                        folders = [x[-1] for x in reader(objs, delimiter=" ")]
```

Comment:
note: connection.list() returns the unwieldy
`[b'(\\HasChildren) "." INBOX', b'(\\HasNoChildren \\Sent) "." INBOX.Sent']`. i'm not thrilled about importing `csv`, but i concede that i don't have a better solution off hand.

Reply:
actually, i'm pretty sure your approach breaks on the second item in my inbox (there's a non-quoted space there).

Reply:
the code here is the same as the one in the `mail` module. I don't know how much it was tested there. 
Can you get the result of the `list()` operation for the item from the server, so that i can try to fix it?